ignore user and domain sync_delay for v2

snopoke opened the pull request:
When looking at this I was also wondering if we need a way to override this for specific users e.g. during training or for 'test' users.

snopoke committed:
uses_reports is a function

snopoke committed:
ignore user and domain sync_delay for v2

emord commented on corehq/apps/app_manager/fixtures/mobile_ucr.py:
```diff
@@ -34,24 +34,27 @@
-        sync_interval = restore_state.project.default_mobile_ucr_sync_interval
+    sync_interval = None
+    if not is_v2:
+        # in v2 sync delay is handled per report in the app
```
I intentionally left this in case we wanted a hard minimum for an app, but I suppose multiple configurations could be confusing. Is that your thought for removing this?

snopoke replied:
yea I think it's a bit confusing but I guess having a hard min could be useful. I'll think about it some more.

emord replied:
Cool. I'm not against this change, so happy to merge once tests pass.

we could also use the `<input min>` attribute to specify a minimum on each input to make it less confusing, although we may want some warning that the minimum is due to an app /domain level setting

snopoke replied:
Went back and forth on this for ages and decided to remove the user and app settings so that it's just the domain and individual report settings.

Nobody is using the user / app settings currently anyway.

snopoke replied:
Still need to make it possible to ignore the sync delay for test users.

emord reviewed (commented):
Yeah. I could see a usecase for overriding this for trainings, although i think for trianings they just use a demo user right? so it's a static restore anyways?

snopoke committed:
remove sync interval from users and apps

snopoke committed:
show domain delay in report module

emord commented on corehq/apps/app_manager/models.py:
```diff
@@ -5284,8 +5284,6 @@ class Application(ApplicationBase, TranslationMixin, HQMediaMixin):
     use_grid_menus = BooleanProperty(default=False)
     grid_form_menus = StringProperty(default='none',
                                      choices=['none', 'all', 'some'])
-    mobile_ucr_sync_interval = IntegerProperty()
```
I've always been hesitant to remove properties like this. If we ever make a new property that has the same name, then these apps could have a value that means something different in the future.

snopoke replied:
True, though I don't think it's worth keeping the code just for that (especially in this case since there are only 3 apps that I know of that have values set for this property).

emord reviewed (commented):
Is this no longer used on ICDS?

snopoke commented:
This isn't needed for ICDS. I checked and all apps either have no value set or the same value as the domain value.

snopoke committed:
fix app aware restore test

This was 1 because the `uses_reports` check wasn't being
executed. See f483a49.

emord reviewed (approved):
Looks good to me. Do you think we will never need this again even with the coming up rollouts before we get servers?

snopoke commented:
I don't think so. We will need a way to ignore the sync delay for test
users but I don't think we'll need the app level settings.
